addons/mrp/models/mrp_bom.py

```diff
@@ -249,7 +249,11 @@ def check_cycle(v, visited, recStack, graph):
             return False
 
         product_ids = set()
-        product_boms = {}
+        cache = self.env.context.get("mrp_explode_cache", {})
+        product_boms = cache.get('product_boms')
+        if product_boms is None:
+            product_boms = {}
```

Comment:
```suggestion
        product_boms = cache.setdefault('product_boms', {}).get('product_boms')
```
less lines

Reply:
Fixed this way
![image](https://user-images.githubusercontent.com/186131/118948236-62c0df80-b958-11eb-825e-586c92915300.png)

-- I don't want to change cache dict, because it would just mean to change new temporary dictionary when context doesn't have the key